test/end-to-end/company.js

```diff
@@ -1933,4 +1933,294 @@ describe("Company endpoint", () => {
             expect(res.body).toHaveProperty("maxOffersReached", true);
         });
     });
+
+    describe("PUT /company/edit", () => {
+        let test_company, test_random, test_offer;
+
+        const test_user_admin = {
+            email: "admin@email.com",
+            password: "password123",
+        };
+
+        const test_user_company = {
+            email: "company@email.com",
+            password: "password123",
+        };
+
+        const test_random_user = {
+            email: "random@email.com",
+            password: "password123",
+        };
+
+        const test_agent = agent();
+
+        beforeEach(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+
+            await Company.deleteMany({});
+
+
+            test_company = await Company.create({
+                name: "Cool Company",
+                contacts: ["112", "122"],
+                bio: "Cool company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            test_random = await Company.create({
+                name: "Random Company",
+                contacts: ["112", "122"],
+                bio: "Random company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            const test_offer_data = generateTestOffer({
+                owner: test_company._id,
+                ownerName: test_company.name,
+                ownerLogo: test_company.logo,
+            });
+
+            test_offer = await Offer.create(test_offer_data);
+
+            await Account.deleteMany({});
+
+            await Account.create({
+                email: test_user_admin.email,
+                password: await hash(test_user_admin.password),
+                isAdmin: true
+            });
+
+            await Account.create({
+                email: test_user_company.email,
+                password: await hash(test_user_company.password),
+                company: test_company._id
+            });
+
+            await Account.create({
+                email: test_random_user.email,
+                password: await hash(test_random_user.password),
+                company: test_random._id
+            });
+        });
+
+        afterEach(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+        });
+
+        afterAll(async () => {
+            await Company.deleteMany({});
+            await Account.deleteMany({});
+        });
+
+        describe("ID Validation", () => {
```

Comment:
Since this is its own describe you could move the login call to a `beforeEach`: prevents code duplication and looks cleaner

Reply:
And add an `afterEach` with a logout action